README.md

```diff
@@ -305,28 +344,151 @@ import { Ziggy } from './ziggy';
 
 Vue.mixin({
     methods: {
-        route: (name, params, absolute) => route(name, params, absolute, Ziggy),
+        route: (name, params, absolute, config = Ziggy) => route(name, params, absolute, config),
     },
 });
 ```
 
-Then you can use the method in your Vue components like so:
+> Note: If you include the `@routes` Blade directive in your views, the `route()` helper will already be available globally, including in your Vue app, so you don't need to import `route` or `Ziggy`. For convenience, you can optionally create a simpler version of the above mixin to make `route()` easily accessibly inside your components:
+>
+> ```js
+> Vue.mixin({ methods: { route });
+> ```
+
+Now you can use the method in your Vue components like so:
 
 ```html
 <a class="nav-link" :href="route('home')">Home</a>
 ```
 
-Thanks to [Archer70](https://github.com/tighten/ziggy/issues/70#issuecomment-369129032) for this solution.
+Thanks to [Scott Christianson](https://github.com/Archer70) for originally sharing [this solution](https://github.com/tighten/ziggy/issues/70#issuecomment-369129032)!
+
+#### React
+
+To use Ziggy with React, start by importing the `route()` function and your Ziggy config. Because the Ziggy config object is not available globally in this setup, you'll have to pass it to the `route()` function manually:
+
+```js
+// app.js
+
+import route from 'ziggy';
+import { Ziggy } from './ziggy';
+
+// ...
+
+route('home', undefined, undefined, Ziggy);
+```
+
+We're working on adding a Hook to Ziggy to make this cleaner, but for now make sure you pass the configuration object as the fourth argument to the `route()` function as shown above.
+
+> Note: If you include the `@routes` Blade directive in your views, the `route()` helper will already be available globally, including in your React app, so you don't need to import `route` or `Ziggy` anywhere.
+
+#### SPAs or separate repos
+
+Ziggy's `route()` helper function is also available as an NPM package, for use in Javascript projects managed separately from their Laravel backend (i.e. without Composer or a `vendor` directory). You can install the NPM package with `npm install ziggy-js`.
+
+To make your routes available on the frontend for this function to use, you can either run `php artisan ziggy:generate` and add the generated routes file to your project, or you can return Ziggy's config as JSON from an endpoint in your Laravel API.
+
+Then, import Ziggy as above:
+
+```js
+// app.js
+
+import route from 'ziggy-js';
+
+import { Ziggy } from './ziggy';
+// or...
+const response = await fetch('/api/ziggy');
+const Ziggy = await response.toJson();
+
+// ...
+
+route('home', undefined, undefined, Ziggy);
+```
+
+The `/api/ziggy` endpoint above that returns Ziggy's config can be something as simple as:
+
+```php
+// routes/api.php
+
+use Tightenco\Ziggy\Ziggy;
+
+Route::get('api/ziggy', fn () => response()->json(new Ziggy));
```

Comment:
I wonder if it would be worth sharing how to do this above when we show how to publish the routes file?